_includes/specials/taproot/en/09-ptlcs.md

```diff
@@ -0,0 +1,153 @@
+and preimage makes it immediately clear that A paid C via
+B.  This can be a significant problem for same-chain and cross-chain
+coinswaps.  Less obviously, this is also a problem for offchain routing
+protocols like LN.  If we imagine a longer routing path where one person
+controls multiple hops along the path, they can see the reuse of
+the same hash and preimage to determine that some nodes are routing
+nodes, increasing the probabability that the remaining nodes are either
+the spender or the receiver.   This is one part of the *linkability
+problem* that may be LN's greatest current privacy weakness.
+
+![Illustration of HTLC linkability problem](/img/posts/2021-07-ln-linkability1.dot.png)
+
+Although [multipath payments][topic multipath payments] partially
+mitigates other aspects of LN's linkability problem, e.g. payment amount
+linkability, it may make worse the hash linkability problem by giving
+surveillance routing nodes more opportunity to correlate hashes.
+
+An additional problem with HTLCs pre-taproot is that their scripts are
+obviously distinct from normal spender scripts.  To a certain degree,
+segwit v0 allowed eliminating this problem by making it possible to use
+presigned transactions free of txid malleability concerns, allowing
+offchain protocols like LN to never publish HTLCs onchain when parties
+are cooperating.  But this adds complexity to some single-shot protocols
+(e.g. cross-chain coinswaps) and it still requires participants use
+somewhat uncommon scripts, such as 2-of-2 multisigs.  This makes it
+easier for surveillants to identify usage
+patterns and, perhaps, make effective guesses about information specific
+to individual users.
+
+### The PTLC solution
+
+In the previous min.sc-style scripts, we had a function that would only
+return true if it was passed a particular value chosen in advance (the
+preimage).  A [signature adaptor][topic adaptor signatures] is similar
+in that it can only be transformed into a valid signature if passed a
+revealed value (the *scalar*).  If we ignore [multisignatures][topic
+multisignature] for the moment, this allows us to transform the HTLC
+scripts from earlier into the following PTLCs:
+
+```hack
+(pk($bob) && pk($alice_adaptor)) || (pk($alice) && older(10))
+(pk($carol) && pk($bob_adaptor)) || (pk($bob) && older(5))
+```
+
+In short, Carol gives Alice the EC *point* for her hidden scalar, Alice
+uses that with a public key she chooses and creates a signature adaptor
+that she gives to Bob; Bob can use the same point with a public key
+he chooses and creates an adaptor he gives to Carol.  Carol reveals the
+scalar by transforming Bob's adaptor into a valid signature, claiming
+Bob's coins.  Bob recovers the scalar from the valid signature, allowing
+him to transform Alice's adaptor into its own valid signature, claiming
+her coins.
+
+This solves the linkability problem against onchain surveillance because
+all anyone looking at the block chain sees are a bunch of valid
```

Comment:
"all anyone" is a bit of an awkward collision. Perhaps:

```suggestion
anyone looking at the block chain only sees a bunch of valid
```

Reply:
Made an edit (no complaints), but can you help me understand what the problem was here?  I'm not sure what "collision" means here.  Does it not sound good to you, coming out like *alanyone* or something?

Reply:
If I read "All John sees are signatures", then it's quite easy to parse. Because "all" and "sees are signatures" are very close together, I can mentally rearrange the sentence to "John only sees signatures" very easily. However, if I replace "John" with "anyone looking at the block chain", then on my first parse I don't know what to attach "all" to. I get as far as "all anyone looking ..." and I get a bit stuck and wonder who "all anyone" is.

It's very obvious what the sentence means on the second parse, but it just interrupted my flow on the first reading. Sorry my first explanation was not very clear.

Reply:
Thanks for the explanation!  **I see** now that the problem is that I broke one of the rules of good writing here: keeping subjects and verbs close together.  I pushed another small edit improve that.